utils/zsh/Makefile

```diff
@@ -20,6 +20,7 @@ PKG_LICENSE:=ZSH
 PKG_LICENSE_FILES:=LICENCE
 PKG_CPE_ID:=cpe:/a:zsh_project:zsh
 
+PKG_FIXUP:=autoreconf
```

Comment:
Why this is required now? In the commit description, there is no such info on this one.

Reply:
from commit:
> use autoconf: git version no longer supports traditional configure script, so it was easier to patch configure.ac

Provided patch is more or less a back port/cherry pick. As zsh's master no longer ships with configure script, instead uses auto-tools, there would had been a huge task of patching configure script that would also had been a huge task for reviewers to check what's changed. Zsh has supported auto-tools for quite some time.

from [docs](https://openwrt.org/docs/guide-developer/packages):

> Packages that are using auto-tools should work with simply “PKG_FIXUP:=autoreconf”.

> Why this is required now? In the commit description, there is no such info on this one.
This is required to use auto-tools over traditional configure script as said in documents.

Reply:
Ok, thanks! Would it be possible to keep this in separate commit to have it clear and simple? This from my point should not be in one huge patch, which would be really hard to track changes.

Reply:
But we are not using master branch and we are using still the same version. Dont you forget to add something?

Reply:
To succeed, we are swapping from traditional configure to configure.ac which makes build process different (equal to master branch minus some bells and whistles not available on release version). It's been like this in autotools version for quite some time, but we haven't utilised it until now.

I could apply another huge patch to fix traditional configure, that would be other way around this, but wouldn't that make things even more complicated? I attempted it in the first place and found out pretty soon that changes required in traditional configure script would be massive and definitely break if it ever gets updated due to amount of changes. So, I sticked to with current autotools version, traditional isn't even supported anymore; whole script has been removed from master branch. Probably been outdated long ago.